tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -1,90 +1,329 @@
+// checks all properties of the instance and destroy the instance and check that resource are deleted.
+func TestAwsE2eSerial(t *testing.T) {
+	setupAWSCloudProvider(t)
+
+	t.Run("test create Windows Instance", testCreateWindowsInstance)
 
-	err = cloud.CreateWindowsVM()
-	assert.NoError(t, err, "error creating Windows instance")
+	t.Run("test destroy Windows instance", testDestroyWindowsInstance)
+}
+
+// testCreateWindowsInstance tests the creation of a Windows instance and checks its properties and attached items.
+func testCreateWindowsInstance(t *testing.T) {
+	setupWindowsInstance(t)
+
+	t.Run("test created instance properties", checkInstanceProperties)
+	t.Run("test instance is attached a public subnet", checkInstanceHasPublicSubnetAndIp)
+	t.Run("test instance has name tag", checkInstanceIsAttachedWithName)
+	t.Run("test instance has infrastructure tag", checkInstanceInfraTagExists)
+	t.Run("test instance is attached the cluster worker's security group", checkInstanceHasClusterWorkerSg)
+	t.Run("test instance is attache a Windows security group", checkInstanceIsAssociatedWithWindowsWorkerSg)
+	t.Run("test instance is associated with cluster worker's IAM", checkInstanceIsAssociatedWithClusterWorkerIAM)
+}
+
+// testDestroyWindowsInstance tests the deletion of a Windows instance and checks if the created instance and Windows
+// security group are deleted.
+func testDestroyWindowsInstance(t *testing.T) {
+	t.Run("test instance is terminated", destroyingWindowsInstance)
+	t.Run("test Windows security group is deleted", checkSgIsDeleted)
+	t.Run("test installer json file is deleted", checkInstallerJsonFileIsDeleted)
+}
+
+// setupAWSCloudProvider creates provider ofr Cloud interface and asserts type into AWS provider.
+// This is the first step of the e2e test and fails the test upon error.
+func setupAWSCloudProvider(t *testing.T) {
+	// The e2e test uses Microsoft Windows Server 2019 Base with Containers image, m4.large type, and libra ssh key.
+	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, awscredentials, "default", dir,
+		imageID, instanceType, sshKey)
 	if err != nil {
-		// TODO: Remove the lingering resources as part of test setup.
-		t.Fatalf("Error while creating a VM delete the virtual machine")
+		assert.FailNow(t, "unable to create an instance")
 	}
-	// Type Assert to AWS so that we can test other functionality
-	aws, ok := cloud.(*awscp.AwsProvider)
+
+	// Type assert to AWS so that we can test other functionality
+	provider, ok := cloud.(*awscp.AwsProvider)
 	if !ok {
-		t.Fatal("Error asserting cloudprovider to aws")
+		assert.FailNow(t, "Error asserting cloudprovider to awsProvider")
 	}
+	awsProvider = provider
+}
+
+// setupWindowsInstance creates a Windows instance and updates global information for infraID, createdInstanceID, and
+// createdSgID. All information updates are required to be successful or instance will be teared down.
+func setupWindowsInstance(t *testing.T) {
```

Comment:
Should we call it as `setUpWindowsInstance` then since we're are creating security group and other things?